docs/data-cookbooks/choice-states.md

```diff
@@ -0,0 +1,46 @@
+# Choice States
+
+Cumulus supports AWS Step Function [`Choice`](https://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-choice-state.html) states. A `Choice` state enables branching logic in Cumulus workflows.
+
+`Choice` state definitions include a list of `Choice Rule`s. Each `Choice Rule` defines a logical operation which compares an input value against a value using a comparison operator. For available comparision operators, review [the AWS docs](https://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-choice-state.html).
+
+If the comparison evaluates to `true`, the `Next` state is followed.
+
+## Example
+
+In [examples/workflow.yml](https://github.com/nasa/cumulus/blob/master/example/workflows.yml) the `ParsePdr` workflow uses a `Choice` state, `CheckAgainChoice`, to terminate the workflow once `isFinished: true` is returned by the `CheckStatus` state.
+
+The `CheckAgainChoice` state definition requires an input object of the following structure:
+
+```json
+{
+  "payload": {
+    "isFinished": false
+  }
+}
+```
+
+Given the above input to the `CheckAgainChoice` state, the workflow would transition to the `PdrStatusReport` state.
+
+```yaml
+    CheckAgainChoice:
+      Type: Choice
+      Choices:
+        - Variable: $.payload.isFinished
+          BooleanEquals: false
+          Next: PdrStatusReport
+        - Variable: $.payload.isFinished
+          BooleanEquals: true
+          Next: StopStatus
+```
+
+## Advanced: Loops in Cumulus Workflows
+
+Understanding the complete `ParsePdr` workflow is not necessary to understanding how `Choice` states work, but `ParsePdr` provides an example of how `Choice` states can be used to create a loop in a Cumulus workflow. 
+
+In the complete `ParsePdr` workflow definition, the state `QueueGranules` is followed by `CheckStatus`. From `CheckStatus` a loop starts: Given `CheckStatus` returns `payload.isFinished: false`, `CheckStatus` is followed by `CheckAgainChoice` is followed by `PdrStatusReport` is followed by `WaitForSomeTime`, which returns to `CheckStatus`. Once `CheckStatus` returns `payload.isFinished: true`, `CheckAgainChoice` proceeds to `StopStatus`.
```

Comment:
I follow this, but I wonder if a supporting image of the workflow would be helpful for users? If so, a reference image for parse-pdr that was used in other parts of the doc lives at `docs/images/sips-parse-pdr.png`.

Reply:
That's a good idea @ifestus

Reply:
Agree, added in https://github.com/nasa/cumulus/pull/523/commits/646b8e9917d4719e42fe2967f4467a50de013e09